lib/matplotlib/backends/backend_qt5.py

```diff
@@ -662,26 +656,27 @@ class NavigationToolbar2QT(NavigationToolbar2, QtWidgets.QToolBar):
 
     def __init__(self, canvas, parent, coordinates=True):
         """coordinates: should we show the coordinates on the right?"""
-        self.canvas = canvas
-        self._parent = parent
         self.coordinates = coordinates
-        self._actions = {}
-        """A mapping of toolitem method names to their QActions"""
-
+        self._actions = {}  # mapping of toolitem method names to QActions.
         QtWidgets.QToolBar.__init__(self, parent)
         NavigationToolbar2.__init__(self, canvas)
 
     @cbook.deprecated("3.3", alternative="self.canvas.parent()")
     @property
     def parent(self):
-        return self._parent
+        return self.canvas.parent()
```

Comment:
I'm not sure this is correct, is it actually guaranteed that the toolbar and the canvas share the same Qt parent?